pkg/api/persistentvolumeclaim/util.go

```diff
@@ -77,5 +77,5 @@ func dataSourceIsEnabled(pvcSpec *core.PersistentVolumeClaimSpec) bool {
 			return true
 		}
 	}
-	return false
+	return true
```

Comment:
You'll need a feature gate rather than just defaulting this to on.  It's somewhat tedious but we currently isolate the DataSources by feature gate, and even a global allow will need its own feature gate.